lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js

```diff
@@ -0,0 +1,255 @@
+* @returns {number} rounded number
+*
+* @example
+* var v = nint( 3.14 );
+* // returns 3.0
+*
+* @example
+* var v = nint( 9.99999 );
+* // returns 10.0
+*/
+function nint( x ) {
+	// TODO: shall be replaced once a dedicated math function is available
+	if ( x >= 0.0 ) {
+		return aint( x + 0.5 );
+	}
+	return aint( x - 0.5 );
+}
+
+
+// MAIN //
+
+/**
+* Estimates the 1-norm of a square, real matrix `A`, reverse communication is used for evaluating matrix-vector products.
+*
+* @private
+* @param {PositiveInteger} N - order of the matrix `A`
+* @param {Float64Array} V - array on final return, `V=A*W`, where `W` is not returned
+* @param {integer} strideV - stride length for `V`
+* @param {NonNegativeInteger} offsetV - starting index of `V`
+* @param {Float64Array} X - input array
+* @param {integer} strideX - stride length for `X`
+* @param {NonNegativeInteger} offsetX - starting index of `X`
+* @param {Int32Array} ISGN - array of signs
+* @param {integer} strideISGN - stride length for `ISGN`
+* @param {NonNegativeInteger} offsetISGN - starting index of `ISGN`
+* @param {Float64Array} EST - array containing estimate of lower bound of 1-norm of `A`
+* @param {NonNegativeInteger} offsetEST - starting index of `EST`
+* @param {Int32Array} KASE - array containing value KASE to control overwritting of `X`
+* @param {NonNegativeInteger} offsetKASE - starting index of `KASE`
+* @param {Int32Array} ISAVE - array for internal storage
+* @param {integer} strideISAVE - stride length for `ISAVE`
+* @param {NonNegativeInteger} offsetISAVE - starting index of `ISAVE`
+* @returns {Int32Array} `ISGN`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Int32Array = require( '@stdlib/array/int32' );
+*
+* var V = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var X = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var ISGN = new Int32Array( [ 1, 1, 1, 1 ] );
+* var EST = new Float64Array( [ 0.0 ] );
+* var KASE = new Int32Array( [ 0 ] );
+* var ISAVE = new Int32Array( [ 0, 0, 0, 0, 0] );
+*
+* ISGN = dlacn2( 4, V, 1, 0, X, 1, 0, ISGN, 1, 0, EST, 0, KASE, 0, ISAVE, 1, 0 );
+* // ISGN => <Int32Array>[ 1, 1, 1, 1 ]
+*/
+function dlacn2( N, V, strideV, offsetV, X, strideX, offsetX, ISGN, strideISGN, offsetISGN, EST, offsetEST, KASE, offsetKASE, ISAVE, strideISAVE, offsetISAVE ) {
+	var altsgn;
+	var estold;
+	var jlast;
+	var tmp;
+	var XS;
+	var i;
+	var j;
+
+	if ( KASE[ offsetKASE ] === 0.0 ) {
+		for ( i = 0; i < N; i++ ) {
+			X[ offsetX + ( i * strideX ) ] = 1.0 / N;
+		}
+		KASE[ offsetKASE ] = 1;
+		ISAVE[ offsetISAVE ] = 1;
+		return ISGN;
+	}
+	if ( ISAVE[ offsetISAVE ] === 1 ) {
+		// First iteration, X has been overwritten by A*X.
+		if ( N === 1 ) {
+			V[ offsetV ] = X[ offsetX ];
+			EST[ offsetEST ] = abs( V[ offsetV ] );
+			KASE[ offsetKASE ] = 0;
+			return ISGN;
+		}
+		EST[ offsetEST ] = dasum( N, X, strideX, offsetX );
+		for ( i = 0; i < N; i++ ) {
+			if ( X[ offsetX + ( i * strideX ) ] >= 0.0 ) {
```

Comment:
Same comment as in other PRs. Use a tmp variable: `ix = offsetX` and then `ix += strideX`. Your implementations consistently have unnecessary flops.